src/util/data.h

```diff
@@ -7,3 +7,5 @@
 
 #define min(a, b) (((a) < (b)) ? (a) : (b))
 #define max(a, b) (((a) > (b)) ? (a) : (b))
```

Comment:
unrelated to this PR but there are some nuances with this, i would use this:
#define MAX(a,b) ({ __typeof__ (a) _a = (a); __typeof__ (b) _b = (b); _a > _b ? _a : _b; })
#define MIN(a,b) ({ __typeof__ (a) _a = (a); __typeof__ (b) _b = (b); _a < _b ? _a : _b; })

Reply:
That compares types, I wanted a simple function to get the min/max value (eg. `min(some_val, 16)`). We can add a min_type/max_type if you want.

Reply:
what do you mean it compares types? this is just to prevent double evaluation, but its fine, it's an edge case